### PR TITLE
Azure pipelines ci

### PR DESCRIPTION
**Pull request by karanotts**
### Change description ###

Refactored all TF code 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: terraform/90-outputs.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+output "service_id" {
+  description = "The ID of the API Management Service created"
+  value       = azurerm_api_management.hmi-apim.id
```
Comment (subrotomukherjee) [inline terraform/90-outputs.tf]: we can  get 'hmi-apim' from the .tfvars as var.prefix just like in other files

File: terraform/24-api-management-product-api.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+resource "azurerm_api_management_product_api" "example" {
+  api_name            = azurerm_api_management_api.hmi-apim-api.name
```
Comment (subrotomukherjee) [inline terraform/24-api-management-product-api.tf]: we can fetch the 'hmi-apim' from the .tfvars as var.prefix just like in other files

File: terraform/10-resource-group.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+resource "azurerm_resource_group" "hmi_apim_rg" {
+    name     = "${var.prefix}-rg-${var.environment}"
```
Comment (subrotomukherjee) [inline terraform/10-resource-group.tf]: As per the naming convention resource group name should end with '-rg'. so we can move '-rg' at the end

Comment (karanotts) [reply terraform/10-resource-group.tf]: Good catch, changed!

File: terraform/90-outputs.tf (hunk 1 above)
Comment (karanotts) [reply terraform/90-outputs.tf]: The code references previously created resource, this way is correct.

File: terraform/24-api-management-product-api.tf (hunk 2 above)
Comment (karanotts) [reply terraform/24-api-management-product-api.tf]: The code references previously created resource, this way is correct.

**Review by subrotomukherjee — APPROVED** [review]

**Pull request closed without merging**